common/transform/src/main/java/ai/h2o/mojos/deploy/common/transform/MojoScorer.java

```diff
@@ -63,14 +86,102 @@ public MojoScorer(
+    ScoreResponse response = scoreResponseConverter.apply(responseFrame, request);
     response.id(pipeline.getUuid());
     return response;
   }
 
+  /**
+   * Method to get shapley values for an incoming request of type {@link ScoreRequest}.
+   *
+   * @param request {@link ScoreRequest}
+   * @return response {@link ContributionResponse}
+   */
+  private ContributionResponse contributionResponse(ScoreRequest request) {
+    MojoFrame requestFrame = scoreRequestConverter
+            .apply(request, pipelineShapley.getInputFrameBuilder());
+    return contribution(requestFrame);
+  }
+
+  /**
+   * Method to get shapley values for an incoming request of type {@link ContributionRequest}.
+   *
+   * @param request {@link ContributionRequest}
+   * @return response {@link ContributionResponse}
+   */
+  public ContributionResponse contributionResponse(ContributionRequest request) {
+    ShapleyType requestedShapleyType = shapleyType(request.getShapleyResults());
+    switch (requestedShapleyType) {
+      case TRANSFORMED:
+        MojoFrame requestFrame = contributionRequestConverter
+                .apply(request, pipelineShapley.getInputFrameBuilder());
+        return contribution(requestFrame);
+      case ORIGINAL:
+        throw new UnsupportedOperationException(UNIMPLEMENTED_MESSAGE);
+      default:
+        throw new IllegalArgumentException(
+                "Only ORIGINAL or TRANSFORMED are accepted enums values "
+                        + "for Shapley results property");
+    }
+  }
+
+  private ContributionResponse contribution(MojoFrame requestFrame) {
+    MojoFrame contributionFrame = doShapleyContrib(requestFrame);
+
+    MojoFrameMeta outputMeta = pipeline.getOutputMeta();
+    ScoringType scoringType = scoringType(outputMeta.getColumns().size());
+    List<String> outputGroupNames = getOutputGroups(outputMeta);
+
+    if (ScoringType.CLASSIFICATION.equals(scoringType)) {
+      return contributionResponseConverter.apply(contributionFrame, outputGroupNames);
+    } else {
+      return contributionResponseConverter.apply(contributionFrame);
+    }
+  }
+
+  private List<String> getOutputGroups(MojoFrameMeta outputMeta) {
+    int numberOutputColumns = outputMeta.getColumns().size();
+    List<String> outputClass =  new ArrayList<>();
+    for (int i = 0; i < numberOutputColumns; i++) {
+      String outputClassName = outputMeta.getColumnName(i);
+      String[] outputClassNameSplit = outputClassName.split("\\.");
+      String refinedOutputClass = outputClassNameSplit[outputClassNameSplit.length - 1 ];
+      outputClass.add(refinedOutputClass);
+    }
+    return outputClass;
+  }
+
+  // note this will be provided by mojo pipeline in the future
+  private ScoringType scoringType(int outputColumnSize) {
+    ScoringType scoringType = null;
+    if (outputColumnSize > 2) {
+      scoringType = ScoringType.CLASSIFICATION;
+    } else if (outputColumnSize == 2) {
```

Comment:
In case you're up for saving a couple of lines and dropping the use of `null`, each conditional could skip setting the field and move straight to `return` the value.